tensorflow/contrib/learn/python/learn/utils/input_fn_utils.py

```diff
@@ -78,16 +80,26 @@ def build_parsing_serving_input_fn(feature_spec, default_batch_size=None):
     feature_spec: a dict of string to `VarLenFeature`/`FixedLenFeature`.
     default_batch_size: the number of query examples expected per batch.
         Leave unset for variable batch size (recommended).
+    base64_encode_example: use base64 to encode serialized examples or not.
 
   Returns:
     An input_fn suitable for use in serving.
   """
   def input_fn():
     """An input_fn that expects a serialized tf.Example."""
-    serialized_tf_example = array_ops.placeholder(dtype=dtypes.string,
-                                                  shape=[default_batch_size],
-                                                  name='input_example_tensor')
-    inputs = {'examples': serialized_tf_example}
+    if base64_encode_example:
+      new_base64_placeholder = tf.placeholder(dtype=tf.string,
```

Comment:
I think `dtypes.string` is `tf.string`, we don't need import `tensorflow` as well.

Reply:
I can change to `dtypes.string` but we still need `tf.decode_base64`.

Reply:
Don't worry, just check API [tf.decode_base64](https://www.tensorflow.org/api_docs/python/tf/decode_base64) and find where the op is: `from tensorflow.python.ops import string_ops` and `string_ops.decode_base64`.

Reply:
Thanks @facaiy and I will update the code to remove the dependency of `tensorflow` package.